exasol_script_languages_developer_sandbox/lib/release_build/run_release_build.py

```diff
@@ -0,0 +1,79 @@
+from exasol_script_languages_developer_sandbox.lib.aws_access.stack_resource import StackResource
+from exasol_script_languages_developer_sandbox.lib.config import ConfigObject
+from exasol_script_languages_developer_sandbox.lib.github_release_access import GithubReleaseAccess
+from exasol_script_languages_developer_sandbox.lib.setup_release_codebuild.release_codebuild import \
+    RELEASE_CODE_BUILD_STACK_NAME
+
+
+def get_environment_variable_override(env_variable: Tuple[str, str]) -> Dict[str, str]:
+    return {"name": env_variable[0], "value": env_variable[1], "type": "PLAINTEXT"}
+
+
+def get_aws_codebuild_project(resources: List[StackResource]) -> StackResource:
+    matching_project = [resource for resource in resources if resource.is_code_build]
+    if len(matching_project) == 0:
+        raise ValueError(f"No release codebuild project deployed. Found following resources: {resources}")
+    if len(matching_project) > 1:
+        raise RuntimeError(f"Multiple release codebuild projects match. Found following matches: {matching_project}")
+    return matching_project[0]
+
+
+def _parse_upload_url(upload_url: str) -> int:
+    """
+    upload_url is expected to have the following format: `https://uploads.github.com/repos/exasol/script-languages-repo/releases/123/assets{?name,label}`
+    where `exasol/script-languages-repo` is the repository for which the release will be created and 123 is the id of the release.
+    This method return release id as integer.
+    """
+    res = re.search(r"^https://uploads.github.com/repos/([a-zA-Z0-9\-_/]+)/releases/([\d]+)/assets", upload_url)
+    if res is None:
+        raise ValueError("Parameter upload_url is in unexpected format.")
+    return int(res.groups()[1])
+
+
+def _execute_release_build(aws_access: AwsAccess, branch: str, asset_id: str,
+                           release_id: int, gh_token: str) -> None:
+    """
+    This function:
+    1. Retrieve resources for the release codebuild stack for that given project
+    2. Find the resource with type CodeBuild
+    3. Creates the environment variables override
+    4. Start and wait for build
+    :raises:
+        `RuntimeError` if build goes wrong or if anything on AWS CodeBuild is not as expected
+        `ValueError` if project is not found on AWS CodeBuild or if the upload is not in expected format.
+    The upload url is only be used to get the release id.
+    """
+    resources = aws_access.get_all_stack_resources(RELEASE_CODE_BUILD_STACK_NAME)
+    matching_project = get_aws_codebuild_project(resources)
+
+    if gh_token is None:
+        raise RuntimeError("Parameter gh_token must not be None.")
+
+    env_variables = [("RELEASE_ID", f"{release_id}"),
+                     ("ASSET_ID", f"{asset_id}"),
+                     ("GITHUB_TOKEN", gh_token)]
+    environment_variables_overrides = list(map(get_environment_variable_override, env_variables))
+    aws_access.start_codebuild(matching_project.physical_id,
```

Comment:
Why not put a dict into start_codebuold for env_variables and let start_codebuold do the transformation

Reply:
I want to keep the implementation in AwsAccess as thin as possible, just a wrapper for boto3, so that I can test the rest.
(Although I have to admit that start_codebuild() did not follow really this guideline).